components/utils.py

```diff
@@ -520,3 +606,282 @@ def untar(source, destination='/tmp', strip=1):
+        :return: The relevant ctx node properties dict.
+        '''
+        ctx_props = self._load_ctx_properties(service_name)
+
+        if write_to_file:
+            # write properties to a local backup path
+            self._write_props_to_file(ctx_props, service_name)
+
+        return ctx_props
+
+    def _write_props_to_file(self, ctx_props, service_name):
+
+        dest_file_path = self._get_props_file_path(service_name)
+        ctx.logger.info('Saving {service} input configuration to {path}'
+                        .format(service=service_name,
+                                path=dest_file_path))
+        write_file(ctx_props, dest_file_path)
+
+    def archive_properties(self, service_name):
+        '''
+        Archive previously used node properties. These properties will be used
+        for rollback purposes. This method should be called ONLY once it's been
+        determined that the node installation has ended and the service is up.
+        :param service_name: The service/node name.
+        '''
+        service_archive_path = os.path.join(self.ROLLBACK_PROPERTIES_PATH,
+                                            service_name)
+        mkdir(service_archive_path)
+        base_service_dir = os.path.join(self.PROPERTIES_PATH, service_name)
+        properties_file_path = os.path.join(base_service_dir,
+                                            self.PROPERTIES_FILE_NAME)
+        ctx.logger.info('Archiving previous inputs for service {service}'
+                        .format(service=service_name))
+        move(properties_file_path, service_archive_path)
+        ctx.logger.info('Setting new properties for service {service}'
+                        .format(service=service_name))
+        move(self._get_props_file_path(service_name), properties_file_path)
+
+    def _get_props_file_path(self, service_name):
+        base_service_dir = os.path.join(self.PROPERTIES_PATH, service_name)
+        if self.is_upgrade:
+            dest_file_path = os.path.join(
+                    base_service_dir,
+                    'upgrade-{0}'.format(self.PROPERTIES_FILE_NAME))
+        else:
+            dest_file_path = os.path.join(base_service_dir,
+                                          self.PROPERTIES_FILE_NAME)
+        return dest_file_path
+
+    def _load_ctx_properties(self, service_name):
+        node_props = dict(ctx.node.properties.get_all())
+        if self.is_upgrade:
+            # Use existing property configuration during upgrade
+            use_existing = node_props.get('use_existing_on_upgrade')
+            if use_existing:
+                install_props_path = os.path.join(self.PROPERTIES_PATH,
+                                                  service_name,
+                                                  self.PROPERTIES_FILE_NAME)
+                ctx.logger.info('Loading existing {service} input config from '
+                                '{path}'.format(service=service_name,
```

Comment:
while I agree that this is more readable (i.e. it looks more like english than {0}), the convention we're using across all scripts is not to explicitly provide the name of the variables. Is  there a reason to break consistency here?